SparkConnector/sparkconnector/configuration.py

```diff
@@ -132,11 +132,16 @@ def configure(self, opts, ports):
 
         # Extend conf adding logging of log4j to java options
         base_extra_java_options = "-Dlog4j.configuration=file:%s" % self.connector.log4j_file
-        extra_java_options = conf.get("spark.driver.extraJavaOptions")
-        if extra_java_options:
-            extra_java_options = base_extra_java_options + " " + extra_java_options
+        extra_java_options = " "
+
+        # SPARK_DRIVER_EXTRA_JAVA_OPTIONS contains the default value of spark.driver.extraJavaOptions.
+        # Later, append the base_java_options. This is done to preserve the default setting and prevent 
+        # base_java_options from completely replacing it. This is necessary e.g. for NXCALS node configuration (customenvs).
+        if os.environ.get("SPARK_DRIVER_EXTRA_JAVA_OPTIONS"):
```

Comment:
I'm thinking about what logic we should follow here. We have three possible extra java options sources:
1. The variable
2. The user configuration via the Spark Connector UI
3. The logging option we need to add

I believe if 1 is defined, it should always be present because otherwise NXCALS can break. 3 also needs to be present always because otherwise the logging breaks. In the case of 2, it is options coming from the user. We could in this case prepend them to ours? So this means that if 1, 2 and 3 exist, the final options are : 2 + 1 + 3. Opinion?